src/measure_theory/function/uniform_integrable.lean

```diff
@@ -0,0 +1,223 @@
+/-
+Copyright (c) 2022 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integral.set_integral
+
+/-!
+# Uniform integrability
+
+This file will be used in the future to define uniform integrability. Uniform integrability
+is an important notion in both measure theory as well as probability theory. So far this file
+only contains the Egorov theorem which will be used to prove the Vitali convergence theorem
+which is one of the main results about uniform integrability.
+
+## Main results
+
+* `measure_theory.egorov`: Egorov's theorem which shows that a sequence of almost everywhere
+  convergent functions converges uniformly except on an arbitrarily small set.
+
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal topological_space
+
+namespace measure_theory
+
+open set filter topological_space
+
+variables {α β ι : Type*} {m : measurable_space α} [metric_space β] {μ : measure α}
+
+section
+
+/-! We will in this section prove Egorov's theorem. -/
+
+namespace egorov
+
+/-- Given a sequence of functions `f` and a function `g`, `not_convergent_seq f g i j` is the
+set of elements such that `f k x` and `g x` are separated by at least `1 / (i + 1)` for some
+`k ≥ j`.
+
+This definition is useful for Egorov's theorem. -/
+def not_convergent_seq (f : ℕ → α → β) (g : α → β) (i j : ℕ) : set α :=
+⋃ k (hk : j ≤ k), {x | (1 / (i + 1 : ℝ)) < dist (f k x) (g x)}
+
+variables {f : ℕ → α → β} {g : α → β}
+
+lemma mem_not_convergent_seq_iff {i j : ℕ} {x : α} : x ∈ not_convergent_seq f g i j ↔
+  ∃ k (hk : j ≤ k), (1 / (i + 1 : ℝ)) < dist (f k x) (g x) :=
+by { simp_rw [not_convergent_seq, mem_Union], refl }
+
+lemma not_convergent_seq_antitone {i : ℕ} :
+  antitone (not_convergent_seq f g i) :=
+λ j k hjk, bUnion_subset_bUnion (λ l hl, ⟨l, le_trans hjk hl, subset.refl _⟩)
+
+lemma measure_inter_not_convergent_seq_eq_zero {s : set α}
+  (hfg : ∀ᵐ x ∂μ, x ∈ s → tendsto (λ n, f n x) at_top (𝓝 (g x))) (i : ℕ) :
+  μ (s ∩ ⋂ j, not_convergent_seq f g i j) = 0 :=
+begin
+  simp_rw [metric.tendsto_at_top, ae_iff] at hfg,
+  rw [← nonpos_iff_eq_zero, ← hfg],
+  refine measure_mono (λ x, _),
+  simp only [mem_inter_eq, mem_Inter, ge_iff_le, mem_not_convergent_seq_iff],
+  push_neg,
+  rintro ⟨hmem, hx⟩,
+  refine ⟨hmem, 1 / (i + 1 : ℝ), nat.one_div_pos_of_nat, λ N, _⟩,
+  obtain ⟨n, hn₁, hn₂⟩ := hx N,
+  exact ⟨n, hn₁, hn₂.le⟩
+end
+
+variables [second_countable_topology β] [measurable_space β] [borel_space β]
+
+lemma not_convergent_seq_measurable_set
+  (hf : ∀ n, measurable[m] (f n)) (hg : measurable g)
+  {i j : ℕ} : measurable_set (not_convergent_seq f g i j) :=
+measurable_set.Union (λ k, measurable_set.Union_Prop $ λ hk,
+  measurable_set_lt measurable_const $ (hf k).dist hg)
+
+lemma measure_not_convergent_seq_tendsto_zero
+  (hf : ∀ n, measurable[m] (f n)) (hg : measurable g)
```

Comment:
That `m` is not necessary: `m` is the default measurable space anyway.
```suggestion
  (hf : ∀ n, measurable (f n)) (hg : measurable g)
```
Can you remove it everywhere else? (except in `not_convergent_seq_measurable_set`, where you would need an `include m` if you remove it)